secure_download/src/routes/upload.js

```diff
@@ -55,24 +52,25 @@ router.post(
   multer({ storage: uploadFileStorage }).single('file'),
   asyncHandler(async (req, res, next) => {
     const {
-      name, uploaded_entity_id, index, checksum, chunk_checksum,
+      name, uploaded_entity_id, upload_path, index, checksum, chunk_checksum,
```

Comment:
Upload API now receives the path where the uploaded content is to be placed for an uploaded dataset.